online-shop-ui/src/components/product-details/product-details.tsx

```diff
@@ -0,0 +1,115 @@
+import { useContext, useEffect, useState } from "react";
+import { ShoppingCartAddContext } from "../../context/shopping-cart-context";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import { Product } from "../../data/products";
+import { productsService } from "../../services/products.service";
+import "./product-details.scss";
+
+const ProductDetails = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [product, setProduct] = useState<Product | null>(null);
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(true);
+
+  const addItemToCart = useContext(ShoppingCartAddContext);
+
+  useEffect(() => {
+    if (!id) return;
+
+    const fetchProduct = async () => {
+      setLoading(true);
+      try {
+        const fetchedProduct = await productsService.findOneById(id);
+        setProduct(fetchedProduct);
+        setError("");
+      } catch (error) {
+        setError("Product not found");
+        console.error("Error fetching product:", error);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchProduct();
+  }, [id]);
+
+  const handleDelete = async () => {
+    if (confirm("Are you sure you want to delete this product?")) {
+      try {
+        await productsService.delete(id!);
```

Comment:
I personally don't like the non-null assertion (!) and I would rather do a check for the id.
Check this out: https://blog.italanta.net/the-case-against-non-null-assertion-operator-ea6848aed636.